GridTariffAPI.lib/Services/TariffPriceCacheDataStore.cs

```diff
@@ -0,0 +1,121 @@
+﻿using GridTariffApi.Lib.Interfaces.External;
+using GridTariffApi.Lib.Models.Holidays;
+using GridTariffApi.Lib.Models.Internal;
+using GridTariffApi.Lib.Models.PriceStructure;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+
+namespace GridTariffApi.Lib.Services
+{
+    public class TariffPriceCacheDataStore : ITariffPriceCacheDataStore
+    {
+        private readonly object _resetLockObject = new object();
+        private TariffPriceStructureRoot _tariffPriceStructureRoot;
+        private IReadOnlyList<Holiday> _holidayRoot;
+        private Dictionary<string, MeteringPointTariff> _meteringPointTariffIndex;
+
+        private DateTimeOffset _tariffCacheValidUntil = DateTime.UtcNow;
+
+        public async Task<IReadOnlyList<MeteringPointTariff>> GetMeteringPointTariffsAsync(List<String> meteringPoints, Func<List<string>, Task<IReadOnlyList<MeteringPointTariff>>> retrieveUncachedMeteringPointTariffsFunc)
+        {
+            var cachedMpTariffs = new Dictionary<string, MeteringPointTariff>();
+            var uncachedMpids = new List<string>();
+
+            foreach (var mpid in meteringPoints)
+            {
+
+                if (_meteringPointTariffIndex.TryGetValue(mpid, out MeteringPointTariff cachedMpTariff))
+                {
+                    cachedMpTariffs[mpid] = cachedMpTariff;
+                }
+                else
+                {
+                    uncachedMpids.Add(mpid);
+                }
+            }
+
+            if (uncachedMpids.Count > 0)
+            {
+                var meteringPointTariffs = await retrieveUncachedMeteringPointTariffsFunc(uncachedMpids);
```

Comment:
Stor forbokstav for retrieveUncachedMeteringPointTariffsFunc ?

Reply:
Nei, det er en Func som tas inn som parameter til metoden, og skal følge parameter-navnekonvensjon med liten forbokstav. Ser litt rart ut, ja :-)